### PR TITLE
Fixes and adjusts RnD

### PR DESCRIPTION
**Pull request by NataKilar**
Fixes and adjusts RnD to work and save properly. Most of the changes here are bug fixes, but there are a few balance and design changes as well.
* Active experiments no longer save. Weakrefs do not save and while it may be feasible to locate the target blueprint per experiment, it's not worth it in my opinion. This also stops players from creating a massive backlog of experiments.
* Invention mode now generates random blueprints a little differently, using weighted probability. This likely needs to be tested and balanced a bit more, but in general it vastly decreases the chance of getting something completely unrelated to what was in the box.
* Fabricators now start with basic recipes, the same as they would in normal rounds. Most of these things are essential for normal gameplay, like pipes and tools.

### CODE REVIEW

File: mods/persistence/modules/science/files/programs/bluprintr.dm [hunk 1]
```diff
@@ -159,7 +159,8 @@
 	data["current_protolathe"] = current_protolathe ? current_protolathe : "Not set"
 
 	var/datum/computer_network/network = computer.get_network()
-	data["blueprint"] = uppertext(open_file ? open_file.recipe.name : "Not set")
+	var/datum/fabricator_recipe/open_recipe = open_file?.get_recipe()
+	data["blueprint"] = uppertext(open_file ? open_recipe.name : "Not set")
```
Comment (Andrew-Fall) [inline mods/persistence/modules/science/files/programs/bluprintr.dm:163]: This can be inlined to data["blueprint"] = uppertext(open_file?.get_recipe().name || "Not set") if you want

File: mods/persistence/modules/science/files/programs/bluprintr.dm [hunk 2]
```diff
@@ -114,16 +114,16 @@
 			to_chat(user, SPAN_NOTICE("Unable to find experiment [href_list["PRG_beginexperiment"]]."))
 			return
 		var/datum/fabricator_recipe/experimental_device/ED = new(experiment.id, experiment.filename)
-		if(!protolathe.try_queue_build(ED, 1))
-			to_chat(user, SPAN_WARNING("Unable to start experiment. A technical error has occured with the protolathe."))
-			QDEL_NULL(ED)
-			return
+		protolathe.try_queue_build(ED, 1)
```
Comment (Andrew-Fall) [inline mods/persistence/modules/science/files/programs/bluprintr.dm:117]: Why was this removed?

Comment (NataKilar) [reply mods/persistence/modules/science/files/programs/bluprintr.dm:117]: try_queue_build() doesn't actually return a value, so this check always failed. Overriding the try_queue_build proc a second time for the sake of this one instance seemed unnecessary to me, but if you think it is, then I can readd the check in some way.

**Review by Andrew-Fall — APPROVED** [review]
